options/options_type.h

```diff
@@ -198,6 +226,44 @@ class OptionTypeInfo {
         verification(_verification),
         flags(_flags) {}
 
+  // Creates an OptionTypeInfo for an enum type.  Enums use an additional
+  // map to convert the enums to/from their string representation.
+  // @param _offset The offset in the options for this enum
+  // @param map The string to enum mapping for this enum
+  template <typename T>
+  static OptionTypeInfo Enum(
+      int _offset, const std::unordered_map<std::string, T>* const map) {
+    return OptionTypeInfo(
+        _offset, OptionType::kEnum, OptionVerificationType::kNormal,
```

Comment:
Better to add comments for each lambda function here to explain what they are used for.